src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,116 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+
+    public static final int INITIAL_SIZE = 10;
+
+    private T[] array = (T[]) new Object[INITIAL_SIZE];
```

Comment:
Don't use raw array casts. Consider using `Arrays.copyOf` or suppressing the unchecked warning if you are sure it's safe.

Reply:
There no other possible to build array type of T.